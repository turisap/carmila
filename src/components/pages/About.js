/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';

const About = () => (
    <div>
        <h1>About page</h1>
    </div>
);

export default About;