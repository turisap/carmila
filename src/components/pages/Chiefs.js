/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';

const Chiefs = () => (
    <div className="container">
        <div className="row">
            <div className="col-md-6">

            </div>
        </div>
    </div>
);

export default Chiefs;