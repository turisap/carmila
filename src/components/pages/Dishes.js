/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';
import { connect } from 'react-redux';
import DishItem from './../DishItem';
import visibleDishes from '../../selectors/visibleDishes';
import getRowsOfDishes from '../../adapters/chunk';
import DishesFilterForm from '../DishesFilterForm';
import { addItem, removeItem, fillCredentials } from '../../actions/orders';


export class Dishes extends React.Component {
    handleAddButton = (item) => {
        this.props.addItemToOrder(item)
    };

    getAddedTitles = () => {
        return [...this.props.items].map(item => item.title);
    };

    render(){
        return (
            <div>
                <section className="dishesFilterForm">
                    <div className="container-fluid">
                        <div className="row">
                            <div className="col-md-12">
                                <DishesFilterForm/>
                            </div>
                        </div>
                    </div>
                </section>
                <section className="dishList">
                    <div className="container-fluid">
                        {getRowsOfDishes(3, this.props.dishes, false, this.handleAddButton, this.getAddedTitles())}
                    </div>
                </section>

            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    dishes : visibleDishes(state.filter.filterBy, state.filter.text),
    items  : state.orders.items
});

const mapDispatchToProps = (dispatch, props) => ({
    addItemToOrder : (item) => dispatch(addItem(item)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Dishes);