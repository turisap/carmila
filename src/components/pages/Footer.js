/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';
import {NavLink} from 'react-router-dom';

const Footer = () => (
    <div className="footer__fix">

        <footer>
            <ul className="footer__links">
                <li><NavLink className="footer__link text-center" to="/about">About Us</NavLink></li>
                <li><NavLink className="footer__link text-center" to="#">Sign Up For News</NavLink></li>
            </ul>
            <ul className="footer__info">
                <li>
                    <p id="footer__info">Our office is located 24, Rd No 2 Krong Battambang.
                        We are open 8a.m. to 5p.m Monday to Friday.</p>
                </li>
            </ul>
        </footer>
    </div>
);

export default Footer;