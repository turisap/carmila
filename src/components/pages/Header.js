/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';

export const Header = (props) => (
    <div>
        <nav className="navbar navbar-expand-lg navbar-light bg-light navbar-right">
            <NavLink className="navbar-brand" to="/">Caldira</NavLink>
            <div className="collapse navbar-collapse" id="navbarColor03">
                <ul className="navbar-nav">
                    <li className="nav-item">
                        <NavLink activeClassName="active" className="nav-link" to="/">Home</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink activeClassName="active" className="nav-link" to="/dishes">Menu</NavLink>
                    </li>
                    {props.haveItemsInOrder &&
                    <li className="nav-item">
                        <NavLink activeClassName="active" className="nav-link" to="/order">Order</NavLink>
                    </li>
                    }
                </ul>
            </div>
        </nav>
    </div>
);

const mapStateToProps = (state) => ({
    haveItemsInOrder : (state.orders.items.length > 0)
});

export default connect(mapStateToProps)(Header);