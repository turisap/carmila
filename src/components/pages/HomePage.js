/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';


const HomePage = () => (
<div>
    <section id="bestFood">
        <div className="container-fluid bestFood__container">
            <div className="row">
                <div className="col-md-6 offset-md-3">
                    <h1 className="text-center bestFood__title">Find the best food in</h1>
                    <h1 className="text-center bestFood__title">Battambang</h1>
                </div>
            </div>
        </div>
    </section>
    <section id="homePagePhotos">
        <div className="container-fluid">
            <div className="row">
                <div className="col-md-4">
                    <img className="homePagePhotos_image homePagePhotos_main" src={"./img/pages/fire.jpg"}/>
                </div>
                <div className="col-md-4">
                    <div className="homePagePhotos_cropper">
                        <img className="homePagePhotos_image"  src={"./img/pages/dessert.jpg"}/>
                        <img className="homePagePhotos_image"  src={"./img/pages/pancakes.jpg"}/>
                    </div>
                </div>
                <div className="col-md-4">
                    <div className="homePagePhotos_cropper">
                        <img className="homePagePhotos_image"  src={"./img/pages/pizza_home.jpg"}/>
                        <img className="homePagePhotos_image"  src={"./img/pages/salami.jpg"}/>
                    </div>
                </div>
            </div>
        </div>
    </section>
</div>
);

export default HomePage;