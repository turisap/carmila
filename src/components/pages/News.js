/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';

const News = () => (
    <div>
        <h1>News</h1>
    </div>
);

export default News;