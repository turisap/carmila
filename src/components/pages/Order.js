/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';
import {connect} from 'react-redux';
import { TimePicker } from 'antd';
import moment from 'moment';
import Link, {LinkedComponent} from 'valuelink';
import {Input} from 'valuelink/tags';
import {NavLink} from 'react-router-dom'
import getRowsOfDishes from '../../adapters/chunk';
import {setPickupTime, fillCredentials, removeItem} from '../../actions/orders';

export class Order extends LinkedComponent {
    state = {
        name : '',
        phone : '',
        timeError : undefined,
        submitted : false
    };

    onDateChange = (e) => {
        if (!e.isAfter(moment().add(1, 'hour'))) {
            this.setState({timeError : 'We need at least one hour to prepare you order'});
            this.props.setPickupTime(undefined);
            return;
        }
        this.setState({timeError : undefined});
        this.props.setPickupTime(e.format('Do MMM, HH:mm'));
    };

    handleOrderPlacing = (e) => {
        e.preventDefault();
        this.setState({submitted:true});
        if(!this.state.timeError) {
            this.props.fillCredentials(this.state)
        }
    };

    render () {
        const linked = this.linkAll();
        const nameLink = Link.state(this, 'name')
            .check(x => x, 'Name is required');
        const phoneLink = Link.state(this, 'phone')
            .check(x => x, 'Phone is required')
            .check(x => x.match(/^[0-9]*$/), 'Phone number should be valid');
        const emptyList = !this.props.orders.items.length > 0;

        return (
            <div>
                <div className={emptyList ? "orderList__holder--empty" : "container-fluid orderList__holder"}>
                    <section className="orderList__heading" style={{visibility: emptyList ? 'hidden' : ''}}>
                        <div className="container-fluid">
                            <h2 className="page-header">We are happy to serve food for you</h2>
                            <hr/>
                        </div>
                    </section>
                    <section className="orderList">
                        {this.props.orders.items && getRowsOfDishes(3, this.props.orders.items, true)}
                    </section>
                    { emptyList &&
                    <div className="orderList__noItems">
                        <h2>You haven't chosen anything yet</h2>
                        <h2 className="error"><NavLink to="/dishes">Take A Look To The Menu</NavLink></h2>
                    </div>
                    }
                    <section className="orderForm">
                        {(this.props.orders.items && this.props.orders.items.length > 0) &&
                        <form onSubmit={this.handleOrderPlacing} className="orderForm__form">
                            <FormInput
                                valueLink={nameLink}
                                submitted={this.state.submitted}
                                className="form-control orderForm__input"
                                placeholder="Your name"
                            />
                            <FormInput
                                valueLink={phoneLink}
                                submitted={this.state.submitted}
                                className="form-control orderForm__input"
                                placeholder="Your phone"
                            />
                            <TimePicker
                                defaultValue={moment().add(1, 'hour')}
                                format={'HH:mm'}
                                onChange={this.onDateChange}
                                size="large"
                            />
                            <button href="#" id="placeOrder" type="submit" className="btn btn-default">
                                Place Order
                            </button>
                        </form>
                        }
                        {(this.state.timeError && this.state.submitted) && <p className="error">{this.state.timeError}</p>}
                    </section>
                </div>
            </div>

        )
    }
}

const FormInput = ({...props}) => (
    <div>
        <Input {...props}/>
        <div className="error-placeholder">
            {(props.valueLink.error && props.submitted) && <p className="error">{props.valueLink.error}</p>}
        </div>
    </div>
);

const mapStateToProps = (state) => ({
    orders : state.orders,
});

const mapDispatchToProps = (dispatch, props) => ({
    setPickupTime : (time) => dispatch(setPickupTime(time)),
    fillCredentials : (data) => dispatch(fillCredentials(data)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Order);




