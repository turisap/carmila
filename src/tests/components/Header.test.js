/**
 * Created by HP on 15-Dec-17.
 */
import React from 'react';
import { shallow } from 'enzyme';
import {Header} from '../../components/pages/Header';

test('should render Header properly', () => {
    const wrapper = shallow(<Header/>);
    expect(wrapper).toMatchSnapshot();
});