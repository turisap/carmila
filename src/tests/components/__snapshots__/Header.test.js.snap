// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header properly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light navbar-right"
  >
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      className="navbar-brand"
      to="/"
    >
      Caldira
    </NavLink>
    <div
      className="collapse navbar-collapse"
      id="navbarColor03"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            to="/"
          >
            Home
          </NavLink>
        </li>
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            to="/dishes"
          >
            Menu
          </NavLink>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
